Sources/StreamChat/Database/DTOs/TeamDTO_Tests.swift

```diff
@@ -0,0 +1,84 @@
+    var database: DatabaseContainerMock!
+
+    override func setUp() {
+        super.setUp()
+        database = try! DatabaseContainerMock(kind: .inMemory)
+    }
+
+    override func tearDown() {
+        AssertAsync.canBeReleased(&database)
+        super.tearDown()
+    }
+
+    func test_teamsForUser_areStoredAndLoadedFromDB() {
+        let teamIds = [UUID().uuidString, UUID().uuidString, UUID().uuidString]
+        let userId = UUID().uuidString
+
+        let payload: UserPayload<NoExtraData> = .dummy(userId: userId, teams: teamIds)
+
+        // Asynchronously save the user payload to the db, which should save teams for user as well.
+        database.write { session in
+            try! session.saveUser(payload: payload)
+        }
+
+        // Asynchronously save the payload to the db
+        database.write { session in
+            try! session.saveUser(payload: payload)
+        }
+
+        // Load the user from the db and check the fields are correct
+        var loadedUserDTO: UserDTO? {
+            database.viewContext.user(id: userId)
+        }
+
+        AssertAsync {
+            Assert.willBeEqual(payload.teams.count, loadedUserDTO?.teams?.count)
+            // It's possible that `loadedUserDTO` is nil so it would sabotage next test.
+            // Let's check this first so we can check channel relationship on next line
+            Assert.willNotBeNil(loadedUserDTO?.teams?.first?.users.first)
+            Assert.willBeEqual(loadedUserDTO?.teams?.first?.users.first, loadedUserDTO)
+        }
+    }
+
+    func test_teamForChannel_isStoredAndLoadedFromDB() {
+        let teamId = UUID().uuidString
+        let channelId = try! ChannelId(cid: "messaging:team")
+
+        let channelDetailPayload: ChannelDetailPayload<NoExtraData> = .dummy(cid: channelId, teamId: teamId)
+
+        let channelPayload: ChannelPayload<NoExtraData> = .init(
+            channel: channelDetailPayload, watcherCount: 0,
+            watchers: nil,
+            members: [],
+            membership: nil,
+            messages: [],
+            channelReads: []
+        )
+
+        // Asynchronously save the channel payload to the db, which should save the team for the channel as well.
+        database.write { session in
+            try! session.saveChannel(payload: channelPayload)
+        }
+
+        // Load the channel from the db and check the team is correct
+        var loadedChannelDTO: ChannelDTO? {
+            database.viewContext.channel(cid: channelId)
+        }
+
+        AssertAsync {
+            Assert.willBeEqual(channelPayload.channel.team, loadedChannelDTO?.team?.id)
+            // It's possible that `loadedChannelDTO` is nil so it would sabotage next test.
+            // Let's check this first so we can check channel relationship on next line
+            Assert.willNotBeNil(loadedChannelDTO?.team?.channels.first)
+            Assert.willBeEqual(loadedChannelDTO?.team?.channels.first, loadedChannelDTO)
+        }
```

Comment:
Just nit: The tests inside `AssertAsync { }` are evaluated in parallel. To achieve what you wanted to achieve, you'd have to do:
```
AssertAsync.willNotBeNil(loadedChannelDTO?.team?.channels.first)
AssertAsync.willBeEqual(loadedChannelDTO?.team?.channels.first, loadedChannelDTO)
```